lib/spack/spack/schema/packages.py

```diff
@@ -72,17 +67,36 @@
                                 'type': 'array',
                                 'default': [],
                                 'items': {'type': 'string'}, }, }, },
-                    'paths': {
-                        'type': 'object',
-                        'default': {},
-                    },
                     'variants': {
                         'oneOf': [
                             {'type': 'string'},
                             {'type': 'array',
                              'items': {'type': 'string'}}],
                     },
+                    'externals': {
+                        'type': 'array',
+                        'items': {
+                            'type': 'object',
+                            'properties': {
+                                'spec': {'type': 'string'},
+                            },
+                            'additionalProperties': True,
+                            'required': ['spec']
+                        }
+                    },
+                    # Deprecated properties, will trigger an error with a
+                    # message telling how to update.
```

Comment:
Can we trigger a deprecation warning instead of an error?

Reply:
I think an error is cleaner. The workflow would be like:
```console
$ spack config get packages
==> Error: /home/culpo/.spack/linux/packages.yaml:3: the attributes "modules" and "paths" in packages.yaml have been deprecated. Run "spack config update packages" to update your configuration

$ spack config update packages
==> File "/home/culpo/.spack/linux/packages.yaml" updated [backup=/home/culpo/.spack/linux/packages.yaml.bkp]
```
and that allows us to deal with a single schema version at a time in the code. Do you see serious drawbacks with this approach?

Reply:
Just backwards compatibility issues. Spack is getting stable enough that people are using it in scripts and expect those scripts to keep working.

Reply:
I'm open to change this, but my opinion is that Spack code base is also getting large enough that we need to strike a balance between the maintenance effort required by some changes and the benefit they bring. I chose this approach considering:
1. That this PR won't affect patch releases of the 0.14 series
2. That people relying on `develop` should expect things to change more often than on a release branch
3. That people likely don't want Spack to change their manually written configuration files silently (so no silent update on first run)

With the current approach I can avoid completely the necessity of dealing with two formats for `packages.yaml` in business code and assume that only the most recent is used (because otherwise validation would have failed at an earlier stage).